compiler/rustc_infer/src/traits/util.rs

```diff
@@ -287,6 +314,14 @@ pub fn transitive_bounds<'tcx>(
     elaborate_trait_refs(tcx, bounds).filter_to_traits()
 }
 
+pub fn transitive_bounds_that_define_assoc_type<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    bounds: impl Iterator<Item = ty::PolyTraitRef<'tcx>>,
+    assoc_name: Ident,
+) -> FxHashSet<ty::PolyTraitRef<'tcx>> {
+    elaborate_trait_refs_that_define_assoc_type(tcx, bounds, assoc_name)
```

Comment:
It seems like we should just inline `elaborate_trait_refs_that_define_assoc_type` into this function. It's never called from anywhere else.